astropy/table/scripts/showtable.py

```diff
@@ -0,0 +1,154 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+"""
+``showtable`` is a command-line script based on ``astropy.io`` and
+``astropy.table`` for printing ASCII, FITS, HDF5 or VOTable files(s) to the
+standard output.
+
+Example usage of ``showtable``:
+
+1. FITS::
+
+    $ showtable astropy/io/fits/tests/data/table.fits
+
+     target V_mag
+    ------- -----
+    NGC1001  11.1
+    NGC1002  12.3
+    NGC1003  15.2
+
+2. ASCII::
+
+    $ showtable astropy/io/ascii/tests/t/simple_csv.csv
+
+     a   b   c
+    --- --- ---
+      1   2   3
+      4   5   6
+
+3. XML::
+
+    $ showtable astropy/io/votable/tests/data/names.xml --max-width 70
+
+               col1             col2     col3  ... col15 col16 col17
+               ---              deg      deg   ...  mag   mag   ---
+    ------------------------- -------- ------- ... ----- ----- -----
+    SSTGLMC G000.0000+00.1611   0.0000  0.1611 ...    --    --    AA
+
+
+
+4. Print all the FITS tables in the current directory::
+
+    $ showtable *.fits
+
+"""
+
+import argparse
+import textwrap
+import warnings
+from astropy import log
+from astropy.table import Table
+from astropy.utils.exceptions import AstropyUserWarning
+
+
+def showtable(filename, args):
+    """
+    Read a table and print to the standard output.
+
+    Parameters
+    ----------
+    filename : str
+        The path to a FITS file.
+
+    """
+    if args.info and args.stats:
+        warnings.warn('--info and --stats cannot be used together',
+                      AstropyUserWarning)
+    if (any((args.max_lines, args.max_width, args.hide_unit, args.show_dtype))
+            and (args.info or args.stats)):
+        warnings.warn('print parameters are ignored if --info or --stats is '
+                      'used', AstropyUserWarning)
+
+    # these parameters are passed to Table.read if they are specified in the
+    # command-line
+    read_kwargs = ('hdu', 'format', 'table_id', 'delimiter')
+    kwargs = {k: v for k, v in vars(args).items()
+              if k in read_kwargs and v is not None}
+    try:
+        table = Table.read(filename, **kwargs)
+        if args.info:
+            print(table.info)
+        elif args.stats:
+            table.info('stats')
+        else:
+            formatter = table.more if args.more else table.pprint
+            formatter(max_lines=args.max_lines, max_width=args.max_width,
+                      show_unit=not args.hide_unit, show_dtype=args.show_dtype)
```

Comment:
The `show_unit` arg is actually a 3-state argument:

- True: always put in a line for the unit
- False: never put in a line for the unit
- None (default): put in a line for the unit if any of the columns has a unit

What is probably wanted here for default is `None`, so you don't get those blank lines in many of the typical cases (like in most of the tests).  So `show_unit=(False if args.hide_unit else None)`.  This doesn't allow for forcing a line for units no matter what, but I think that is fine.

Reply:
Indeed, I wanted `None` as default, thanks.